src/stores/userImage.ts

```diff
@@ -0,0 +1,288 @@
+      // 条件canvasで使用されていないかつToolbarで非表示
+      return uploadedImage.countOnCanvas <= 0 && !uploadedImage.show
+    },
+
+    // ログイン時に呼び出し、削除判定
+    // countOnCanvas === 0 && show === false ならfireStoregeとfirebaseから画像削除
+    async deleteImageFromStorageWithLogin() {
+      const { userImageStorage } = this
+      if (userImageStorage !== undefined) {
+        // eslint-disable-next-line no-restricted-syntax
+        for (const key of Object.keys(userImageStorage)) {
+          const userImage: UploadedImage | undefined = userImageStorage[key]
+          if (
+            userImage !== undefined &&
+            this.isDeleteImageFromStorage(userImage)
+          ) {
+            // Storageから削除
+            this.deleteImageFromStorage(userImage)
+            // firestoreから削除
+            // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+            delete userImageStorage[key]
+          }
+        }
+
+        const { uid } = storeToRefs(useAuthStore())
+        const docRef = doc(db, 'userImageStorage', uid.value)
+        await setDoc(docRef, userImageStorage)
+      }
+    },
+
+    // Storageからユーザーがアップロードした画像を削除する
+    deleteImageFromStorage(userImage: UploadedImage) {
+      const deletedRef = ref(
+        storage,
+        `user-image/${userImage.userUid}/user-upload/${userImage.id}.${userImage.fileExtension}`,
+      )
+      deleteObject(deletedRef)
+        .then(() => {
+          console.log('image deleted')
+        })
+        .catch((error) => {
+          console.log(error.code)
+        })
+    },
+
+    // 保存前のKonvaImages[]と保存時のKonvaImages[]を受け取る -> hashmap {imageId: 2, imageId:1}
+    diffHashMap(
+      firstCanvasImages: KonvaImage[],
+      savedCanvasImages: KonvaImage[],
+    ) {
+      const hashmap = new Map<string, number>()
+
+      // 保存時のキャンバス上の各画像のimageIdに対してcountを1増やす
+      for (let i = 0; i < savedCanvasImages.length; i += 1) {
+        const { imageId } = savedCanvasImages[i]
+        const count = hashmap.get(imageId) ?? 0
+        hashmap.set(imageId, count + 1)
+      }
+
+      // 保存前のキャンバス上の各画像のimageIdに対してcountを1減らす
+      for (let i = 0; i < firstCanvasImages.length; i += 1) {
+        const { imageId } = firstCanvasImages[i]
+        const count = hashmap.get(imageId) ?? 0
+        hashmap.set(imageId, count - 1)
+      }
+
+      // 差分のハッシュマップを作成する
+      const diff = new Map<string, number>()
+      // eslint-disable-next-line no-restricted-syntax
+      for (const [key, value] of hashmap) {
+        if (value !== 0) {
+          diff.set(key, value)
+        }
+      }
+      return diff
+    },
+
+    async saveImageCountToFirebase(
+      firstCanvasImages: KonvaImage[],
+      savedCanvasImages: KonvaImage[],
+    ) {
+      const { userImageStorage } = this
+
+      const countHashmapDiff = this.diffHashMap(
+        firstCanvasImages,
+        savedCanvasImages,
+      )
+      console.log(countHashmapDiff)
+
+      if (userImageStorage !== undefined) {
```

Comment:
`countHashmapDiff`変数はifの中でしか使っていなので中に入れて処理短縮を

```suggestion
      if (userImageStorage !== undefined) {
           const countHashmapDiff = this.diffHashMap(
             firstCanvasImages,
             savedCanvasImages,
            )
```